exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/transactions/TransferTxHistoryTest.java

```diff
@@ -66,13 +67,22 @@ void transfersHistoryBetweenTwoAccountsTest() throws Exception {
       // Create and execute 1st transaction
       long seed1 = 1L;
       long transferSum1 = 40L;
-      TransferTx tx1 = withMockMessage(seed1, ACCOUNT_1, ACCOUNT_2, transferSum1);
-      tx1.execute(view);
+      TransferTx tx1 = new TransferTx(seed1, ACCOUNT_2, transferSum1);
+      TransactionContext context1 = newContext(view)
+          .withTxMessageHash(MESSAGE_HASH)
+          .withAuthorKey(ACCOUNT_1)
+          .create();
+      tx1.execute(context1);
+
       // Create and execute 2nd transaction
       long seed2 = 2L;
       long transferSum2 = 10L;
-      TransferTx tx2 = withMockMessage(seed2, ACCOUNT_2, ACCOUNT_1, transferSum2);
-      tx2.execute(view);
+      TransferTx tx2 = new TransferTx(seed2, ACCOUNT_1, transferSum2);
+      TransactionContext context2 = newContext(view)
+          .withTxMessageHash(MESSAGE_HASH)
```

Comment:
I would not use the same hash for different transactions in an IT because it is not possible.